### DISCUSSION TITLE
URLs Needed to Install

### DISCUSSION THREAD

**Discussion by AhmedAbuElyazeed** [apache/airflow#28115] [Q&A]
Greetings,

I'm trying to install Apache-Airflow on an offline server that has proxy. I tried installing it on another machine with internet connection and copy-paste the whole Anaconda environment, but it didn't work.

I can request some URLs to be allowed through the proxy so how can I know exactly which URLs are needed to install Apache-Airflow and all its dependencies on rhel.

**Comment by potiuk** [comment]
There is no such inventory available. You would need to log all requests on the proxy and enable them. But instead you can simply get all the .whl files and install them locally. Don't use `anaconda` for that - or at least if you do, you need to figure it out ourselves - Airlfow is really (see the documentation) only supported for `pip` installation and you should use `pip` . 

If you use `pip`, you can use some inspirations friom https://airflow.apache.org/docs/docker-stack/build.html#build-images-in-security-restricted-environments. It describes how to build images in security restricted environments but part of the process (exporting .whl files that you can use for such installation) is applicable to your case.

**Reply by potiuk** [reply]
Those commands (you have to adjust it for your needs of cours) is the key part:
```
curl -Lo "docker-context-files/constraints-3.7.txt" \
    "https://raw.githubusercontent.com/apache/airflow/constraints-${AIRFLOW_VERSION}/constraints-3.7.txt"

pip download --dest docker-context-files \
    --constraint docker-context-files/constraints-3.7.txt  \
    "apache-airflow[async,celery,elasticsearch,kubernetes,postgres,redis,ssh,statsd,virtualenv]==${AIRFLOW_VERSION}"
````
